src/generics.rs

```diff
@@ -845,8 +845,8 @@ pub(crate) mod parsing {
     #[cfg_attr(docsrs, doc(cfg(feature = "parsing")))]
     impl Parse for TraitBound {
         fn parse(input: ParseStream) -> Result<Self> {
-            let modifier: TraitBoundModifier = input.parse()?;
             let lifetimes: Option<BoundLifetimes> = input.parse()?;
+            let modifier: TraitBoundModifier = input.parse()?;
```

Comment:
I don't think this change is correct. As far as I know, there have only ever been 2 kinds of rustc: one where `?for<'a> Trait` is valid trait bound syntax (all versions prior to 1.82) and one where neither `?for<'a> Trait` nor `for<'a> ?Trait` is valid trait bound syntax. There has never been any version of Rust where the syntax implemented here is valid trait bound syntax.

Could you provide some more justification about this change?